src/model.jl

```diff
@@ -66,21 +66,68 @@ function _make_bus_demand_flexibility_amount(
     sets::Sets,
 )::Tuple{Matrix,Matrix}
 
-    # Demand flexibility up profile
-    simulation_demand_flex_amt_up = Matrix(
+    # combine two profiles with user input overriding DOE flexibility
+    simulation_demand_flex_amt_up = zeros(end_index - start_index + 1, sets.num_bus)
+    simulation_demand_flex_amt_dn = zeros(end_index - start_index + 1, sets.num_bus)
+    
+    # reformat flexibility cost up/dn to appropriate dimensions
+    new_demand_flex_cost_up = zeros(size(demand_flexibility.cost_up, 1), sets.num_bus)
+    new_demand_flex_cost_dn = zeros(size(demand_flexibility.cost_dn, 1), sets.num_bus)
```

Comment:
I think we need to account for the possibility that `demand_flexibility.cost_up` and `demand_flexibility.cost_dn` are `nothing`. Maybe keep all the cost-related code together and have an `if` statement to check?